monitor/internal/pod/delete_controller.go

```diff
@@ -142,16 +143,19 @@ func deleteControllerProcessItem(backgroundCtx context.Context, c client.Client,
 	// For StatefulSets we need to account for another special case: pods that move between nodes *keep* the same UID, so they won't fit the check below.
 	// However, we can simply double-check the node name in the same way how we already filter events in the watcher/monitor
 	if pod.Spec.NodeName != nodeName {
-		zap.L().Debug("DeleteController: the pod is now on a different node, send destroy event", zap.String("puID", podUID), zap.String("namespacedName", req.String()), zap.String("podNodeName", pod.Spec.NodeName), zap.String("nodeName", nodeName))
-		if err := pc.Policy.HandlePUEvent(
-			ctx,
-			podUID,
-			common.EventDestroy,
-			policy.NewPURuntimeWithDefaults(),
-		); err != nil {
-			// we don't really care, we just warn
-			zap.L().Warn("DeleteController: Failed to handle destroy event", zap.String("puID", podUID), zap.String("namespacedName", req.String()), zap.Error(err))
-		}
+		// NODE REBOOT can make us land here.
+		// 1. Don't send a delete event as we know that the pod is present in cluster.
```

Comment:
We must delete the PU. PUs are bound to the enforcers that manage them. So even if the same pod moved to a new node, this must entail a PUCreate and PUStart on the other host, and a PUDestroy on this node.